python/nav/portadmin/snmp/base.py

```diff
@@ -184,10 +184,14 @@ def get_if_alias(self, if_index):
         return safestring(self._query_netbox(self.IF_ALIAS_OID, if_index))
 
     def get_all_if_alias(self):
-        """Get all aliases for all interfaces."""
-        return [
-            (oid, safestring(value)) for oid, value in self._bulkwalk(self.IF_ALIAS_OID)
-        ]
+        """Get all aliases for all interfaces.
+
+        :returns: A dict describing {ifIndex: ifAlias}
+        """
+        return {
+            OID(oid)[-1]: safestring(value)
+            for oid, value in self._bulkwalk(self.IF_ALIAS_OID)
+        }
```

Comment:
This needs a standing ovation meme-gif. Remember life before dict comprehensions? =)